src/Morphir/Scala/Feature/Codec.elm

```diff
@@ -0,0 +1,508 @@
+                                , valueType =
+                                    Just
+                                        (Scala.TypeApply
+                                            (Scala.TypeRef [ "io", "circe" ] "Decoder")
+                                            [ Scala.TypeRef scalaTypePath (scalaName |> Name.toTitleCase)
+                                            ]
+                                        )
+                                , value =
+                                    encodeValue |> Scala.Lambda [ ( "c", Just (Scala.TypeRef [ "io.circe" ] "HCursor") ) ]
+                                }
+                            )
+                        ]
+                    )
+
+        Type.CustomTypeDefinition typeParams accessControlledCtors ->
+            Ok
+                (mapCustomTypeDefinitionToDecoder
+                    currentPackagePath
+                    currentModulePath
+                    accessControlledModuleDef.value
+                    typeName
+                    typeParams
+                    accessControlledCtors
+                )
+
+
+mapCustomTypeDefinitionToDecoder : Package.PackageName -> Path -> Module.Definition ta (Type ()) -> Name -> List Name -> AccessControlled (Type.Constructors a) -> List (Scala.Annotated Scala.MemberDecl)
+mapCustomTypeDefinitionToDecoder currentPackagePath currentModulePath moduleDef typeName typeParams accessControlledCtors =
+    let
+        ( scalaTypePath, scalaName ) =
+            ScalaBackend.mapFQNameToPathAndName (FQName.fQName currentPackagePath currentModulePath typeName)
+
+        patternMatch : List ( Scala.Pattern, Scala.Value )
+        patternMatch =
+            accessControlledCtors.value
+                |> Dict.toList
+                |> List.map
+                    (\( ctorName, ctorArgs ) ->
+                        composeDecoder ctorName ctorArgs
+                    )
+    in
+    [ Scala.withoutAnnotation
+        (Scala.ValueDecl
+            { modifiers = [ Scala.Implicit ]
+            , pattern = Scala.NamedMatch ("decode" :: typeName |> Name.toCamelCase)
+            , valueType = Just (Scala.TypeRef [] (typeName |> Name.toTitleCase))
+            , value =
+                Scala.Lambda [ ( "c", Just (Scala.TypeRef [ "io", "circe" ] "HCursor") ) ]
+                    (Scala.Match (Scala.Variable (scalaName |> Name.toTitleCase)) (Scala.MatchCases patternMatch))
+            }
+        )
+    ]
+
+
+
+--( "c", Just (Scala.TypeRef [ "io", "circe" ] "HCursor") )
+
+
+composeDecoder : Name -> List ( Name, Type ta ) -> ( Scala.Pattern, Scala.Value )
+composeDecoder ctorName ctorArgs =
+    let
+        args =
+            ctorArgs
+                |> List.map
+                    (\( argName, argType ) ->
+                        let
+                            typeRef : Scala.Value
+                            typeRef =
+                                genEncodeReference argType |> Result.withDefault (Scala.Variable "")
```

Comment:
Empty variable?

Reply:
This has been fixed by pattern matching Ok and Err